internal/engine/configs/trigger_queue_subscriber.go

```diff
@@ -70,3 +76,26 @@ func (s *TriggerQueueSubscriber) OnChange(ctx context.Context, st store.RStore,
 	s.lastUpdate = &obj
 	return nil
 }
+
+func removeDisabledManifests(ctx context.Context, st store.RStore) {
+	state := st.RLockState()
+	defer st.RUnlockState()
+
+	for _, mn := range state.TriggerQueue {
+		if uir, ok := state.UIResources[string(mn)]; ok {
+			if uir.Status.DisableStatus.DisabledCount > 0 {
+				spanID := logstore.SpanID(fmt.Sprintf("unqueue:%s", mn))
+				msg := fmt.Sprintf("Will not build resource %q: it is disabled", mn)
+				st.Dispatch(store.NewLogAction(mn, spanID, logger.InfoLvl, nil, []byte(msg)))
+
+				st.Dispatch(RemoveFromTriggerQueueAction{Name: mn})
+			}
+		}
+	}
+}
+
+type RemoveFromTriggerQueueAction struct {
+	Name model.ManifestName
```

Comment:
shouldn't this be handled in the reducer loop? it seems weird for this state to bounce out to a subscriber, which then dispatches an action back into the reducer.

Reply:
yeah, that makes more sense